eos-link-host-mali-driver

```diff
@@ -22,14 +22,75 @@
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 
 EXTENSION_REF=org.freedesktop.Platform.GL.host/arm/1.6
-MALI_DIR=/usr/lib/mali400/arm-linux-gnueabihf
 
-DEST_DIR="/var/lib/flatpak/extension/$EXTENSION_REF"
-mkdir -p $(dirname "$DEST_DIR")
-ln -sf "$MALI_DIR" "$DEST_DIR"
+#
+# create_flatpak_gl_host_extension FLATPAK_DIR MALI_DRIVER
+#
+# Creates a flatpak GL host extension in $FLATPAK_DIR from $MALI_DRIVER
+#
+function create_flatpak_gl_host_extension {
+    FLATPAK_DIR=$1
+    MALI_DRIVER=$2
+
+    if test ! -d $FLATPAK_DIR; then
+        return;
+    fi
+
+    DEST_DIR=$FLATPAK_DIR/extension/$EXTENSION_REF
+
+    # Ensure destination directory
+    mkdir -p $DEST_DIR
+
+    # Create lib link for flatpak GL extension
+    ln -sr $DEST_DIR $DEST_DIR/lib
 
-if test -d /var/endless-extra/flatpak; then
-    DEST_DIR="/var/endless-extra/flatpak/extension/$EXTENSION_REF"
-    mkdir -p $(dirname "$DEST_DIR")
-    ln -sf "$MALI_DIR" "$DEST_DIR"
+    # Hardlink mali driver
+    ln -t $DEST_DIR $MALI_DRIVER
+
+    # Create mali driver wrapper with missing symbol
+    # (Binary generated from libMaliWrapper.c)
+(   base64 -d <<EOF
```

Comment:
Can't we ship this in the ostree as well, through the mali package?

Reply:
I think the question is: do we want this change only in the runtime, or is it fine to have it in the host? If the later, then we should probably ship it in the ostree, and in that case I believe you could drop all the other changes in this file.